src/main/java/homeworks/Infrastructure/data/FileDataMapper.java

```diff
@@ -0,0 +1,96 @@
+package homeworks.Infrastructure.data;
+
+import java.io.*;
+
+public class FileDataMapper implements DataMapper {
+    User[] users = new User[9];
+
+    public FileDataMapper() {
+        try (BufferedReader reader = new BufferedReader(new FileReader("users.txt"))){
+            String rb = null;
+            while ((rb = reader.readLine()) != null) {
+                String[] splitedUser = rb.split(";");
+
+                for (int i = 0; i < users.length; i++) {
+                    if (users[i] == null) {
+                        users[i] = new User(splitedUser[0], splitedUser[1], splitedUser[2], splitedUser[3]);
+                        break;
+                    }
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    class reader implements AutoCloseable {
+
+        @Override
```

Comment:
not necessary implements  AutoCloseable  interface, it already implemented in Buffered reader
![image](https://user-images.githubusercontent.com/19190298/96043286-6139b100-0e77-11eb-83a3-97505b2cb080.png)